lib/ble.toit

```diff
@@ -257,23 +270,157 @@ class Client:
     state := resource_state_.wait_for_state STARTED_EVENT_
     resource_state_.clear_state STARTED_EVENT_
 
+class ServerConfig:
+  resource_group_ := null
+  services/List := []
+
+  constructor:
+    resource_group_ = ble_server_config_init_
+
+  add_service uuid -> Service:
+    service := Service resource_group_ uuid
+    services.add service
+    return Service resource_group_ uuid
+
+
+/** 
+Defines a BLE service with characteristics 
+*/
+class Service:
+  /**
+  The UUID of the service.
+  
+  For 16 and 32 bit UUIDs, form the BLE variant with the top-level uuid function.
+  */
+  uuid/uuid_pkg.Uuid
+  
+  resource_group_/any
+  resource_/any
+
+  characteristics_/Map := {:}
+
+  constructor .resource_group_ .uuid:
+    resource_ = ble_add_server_service_ resource_group_ uuid.to_byte_array
```

Comment:
Looks like we will be leaking the resource as long as the resource group is around. In this case, that might not be so bad because it sticks to the ServerConfig that doesn't allow unregistering.

Reply:
Yes. Services will be kept around as long as the ServerConfiguration is around. There is no way to deregister a single service in nimble AFAIK.